### PR TITLE
Add subtitle addon type

### PR DESCRIPTION
**Pull request by razzeee**
Please review.

`subtitles.py` done by @phate89 
I feel like there is still some stuff to get rid off. So let's talk about that.

 And I'm not sure if I should have replaced some of the stuff with placeholders?

### CODE REVIEW

**Comment by phate89** [comment]
You mean to clean the sample? most of the code is to retrieve info from the playing. I looked to other addons and they all use almost exactly this code....so  I think it's good to have it into provide something ready.
We could use json to retrieve the info needed but the cleaning code is still needed

**Comment by razzeee** [comment]
Well I just thought it might scare people away that just started doing addons. 
But if you think it's worth it, I'll start to add some more code comments in the next days. Feel free to provide me with some more if you like.  Here's hoping that there are really comments I could add ;)

**Comment by phate89** [comment]
Subtitle addons are really basic stuff and they just have to respond to specific parameters.
if you open the addon with `plugin://script.service.foo/?action=search` or `plugin://script.service.foo/?action=manualsearch&searchstring=usersearchedstring` the plugin should return a list of fileitem with a link to the module to download the subtitle (for example `plugin://script.service.foo/?action=download&id=1`).
Then it will have to handle the call of `plugin://script.service.foo/?action=download&bla=bla` downloading the subtitle and returning a fileitem with url the local path of the downloaded subtitle

I understand that it's a lot of code. we could drop the code to gather info and download the subtitle but it's something that ALL subtitle addons need to chose what subtitle display and to save the file..

The alternative is to use/create some extra module to hide this so that all subtitle addons might use..

**Comment by razzeee** [comment]
Well let's wait for some third party people like @MartijnKaijser @phil65 or @enen92 reviewing this.
I'm fine with using it as is.

File: generators/app/templates/subtitle.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import urllib2, sys, urlparse, urllib, os, unicodedata
```
Comment (razzeee) [inline generators/app/templates/subtitle.py]: Add file encoding `utf-8`?

Comment (phil65) [reply generators/app/templates/subtitle.py]: yup, + pep8 says separate lines for each import.

File: generators/app/templates/subtitle.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import xbmcgui, xbmcplugin, xbmc, xbmcaddon, xbmcvfs
+
+ADDON = xbmcaddon.Addon()
+SCRIPT_ID   = ADDON.getAddonInfo('id')
+PROFILE = xbmc.translatePath(ADDON.getAddonInfo('profile'))
+TEMP = os.path.join( PROFILE, 'temp', '')
+HANDLE=int(sys.argv[1])
```
Comment (phil65) [inline generators/app/templates/subtitle.py]: spaces are wrong here on some occasions.

File: generators/app/templates/subtitle.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+if not xbmcvfs.exists(TEMP):
+  xbmcvfs.mkdirs(TEMP)
+
+#function to retrieve parameters in a dictionary
+def getParams():
+    if len(sys.argv) > 2:
+        return dict(urlparse.parse_qsl(sys.argv[2].lstrip(b'?')))
```
Comment (phil65) [inline generators/app/templates/subtitle.py]: why that "b" prefix?

File: generators/app/templates/subtitle.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    return {}
+
+def normalizeString(str):
+    return unicodedata.normalize(
+        'NFKD', unicode(unicode(str, 'utf-8'))
+    ).encode('ascii', 'ignore')
+
+def getInfo():
+    item = {}
+    item['temp']               = False
```
Comment (phil65) [inline generators/app/templates/subtitle.py]: no vertical alignment.

File: generators/app/templates/subtitle.py (hunk 1 above)
Comment (phate89) [reply generators/app/templates/subtitle.py]: yes I deleted utf-8 by mistake

File: generators/app/templates/subtitle.py (hunk 3 above)
Comment (phate89) [reply generators/app/templates/subtitle.py]: b is 3.x stuff.. in 2.x is pointless and we can remove it

**Comment by BigNoid** [comment]
Subtitles.py uses camelCase, PascalCase and the pep-8 recommendation of lowercase and underscores mixed together for functions. I think we should provide a consistent base in that regard if this is to be a template.

File: generators/app/templates/subtitle.py (hunk 3 above)
Comment (razzeee) [reply generators/app/templates/subtitle.py]: so just loose the b?
`.lstrip('?')))`

Comment (phate89) [reply generators/app/templates/subtitle.py]: yes it's just to remove the question mark from `?bla=foo&..`

**Comment by razzeee** [comment]
Run Pep8 on the Subtitles.py file, all errors fixed apart from 2-3 lines too long.

**Comment by razzeee** [comment]
I would like to merge this.

@phate89 could you do a followup PR with some more code comments? I feel like I understand to less about some of the stuff set here.

**Comment by phate89** [comment]
@Razzeee yes sure.. anyway what code style we want to use for variables/functions? to have a single one everywhere

**Comment by razzeee** [comment]
@phil65 should answere that, he's far more into the python stuff

**Comment by MartijnKaijser** [comment]
Follow https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions

**Pull request merged**